### ISSUE TITLE
Autocomplete: isSelected(option) doesn't work correctly when 'multiple' is true

### ISSUE THREAD

**Issue by bogdan0083** [primefaces/primevue#5435]
### Describe the bug

When we have `multiple` set to `true` items in content list don't get properly selected:

* `aria-selected` for every item always `false` even when multiple items are selected
* `data-p-highlight`  for every item always `false` even when multiple items are selected
* `context.selected` in `PassThrough` options is always `false` for every item even when item is selected 

It happens because `isSelected(option)` method internally compares `modelValue` and our option, but when `multiple` is set to `true` modelValue is an array:
```js
        isSelected(option) {
            // modelValue is an array, this will always be false
            return ObjectUtils.equals(this.modelValue, this.getOptionValue(option), this.equalityKey);
        },
```

### Reproducer

https://stackblitz.com/edit/nwuse1?file=src%2FApp.vue

### PrimeVue version

3.51.0-SNAPSHOT

### Vue version

3.x

### Language

ALL

### Build / Runtime

Nuxt

### Steps to reproduce the behavior

1. Open my reproduction link
2. Initialize the project
3. Type `s` in input field
4. Choose `s-4`
5. Type `s` again
6. Now check selected item in content list: `aria-selected` and `data-p-highlight` attributes are set to `false`

### Expected behavior

Selected item in content list should have `aria-selected` and `data-p-highlight` set to true. `PassThrough` option of `context.selected` should also be true.

**Comment by bogdan0083** [comment]
I created a PR that fixes this issue: https://github.com/primefaces/primevue/pull/5436

**Comment by bogdan0083** [comment]
@tugcekucukoglu Your solution is fine but now when `multiple` is `false` and you type something in input it throws error: 

```
Uncaught (in promise) TypeError: (intermediate value).some is not a function
    NuxtJS 9
        isSelected
        item
        getItemValue
        _getOptionValue
        cx
        content
        renderList
        content
        renderFnWithContext
```
<img width="619" alt="Снимок экрана 2024-03-19 в 18 24 49" src="https://github.com/primefaces/primevue/assets/7077307/71fbc9c8-0816-4e8b-a635-413cecce3e59">

You can open autocomplete docs on /autocomplete/#basic and confirm this.
